packages/material-ui/src/Select/SelectInput.js

```diff
@@ -15,17 +15,14 @@ import useForkRef from '../utils/useForkRef';
 import useControlled from '../utils/useControlled';
 import selectClasses, { getSelectUtilitiyClasses } from './selectClasses';
 
+// TODO: enable overrides for the icon and nativeInput slots via the components prop
```

Comment:
Now when I think about this, we could do what we've done in https://github.com/mui-org/material-ui/pull/25777

Reply:
This sounds like a great idea

Reply:
Let me open an issue

Reply:
Here it is https://github.com/mui-org/material-ui/pull/25796

Reply:
I guess we will rebase this PR on #25796 once merged.

Reply:
Yeah, makes sense 👍

Reply:
Rebased